easybuild/tools/github.py

```diff
@@ -958,6 +958,55 @@ def not_eligible(msg):
     return res
 
 
+def close_pr(pr, reason):
+    """
+    Close specified pull request
+    """
+    github_user = build_option('github_user')
+    if github_user is None:
+        raise EasyBuildError("GitHub user must be specified to use --close-pr")
+
+    pr_target_account = build_option('pr_target_account')
+    pr_target_repo = build_option('pr_target_repo')
+
+    pr_url = lambda g: g.repos[pr_target_account][pr_target_repo].pulls[pr]
+    status, pr_data = github_api_get_request(pr_url, github_user)
+    if status != HTTP_STATUS_OK:
+        raise EasyBuildError("Failed to get data for PR #%d from %s/%s (status: %d %s)",
+                             pr, pr_target_account, pr_target_repo, status, pr_data)
```

Comment:
grabbing `pr_data` is becoming pretty common, we should introduce a `fetch_pr_data` function that we can call where relevant, code duplication is indeed becoming a bit much in `github.py`...